### PR TITLE
Endowment claiming

### PR DESCRIPTION
**Pull request by ap-justin**
Closes BG-1063
Towards BG-1200

## Explanation of the solution
* https://github.com/AngelProtocolFinance/devops/pull/535
* unclaimed endow profile now has claim links
image

* registering via endow profile claim link, or opting to switch from fresh to claim (in documentation),  registration context would appear

image

* application list have `claim/new` type
image

* application details claim tooltip link (points to profile)
image

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
-

## UI changes for review

When major UI changes are introduced with a PR, please include links to URLS to compare or screenshots demonstrating the difference and notify on design changes

### CODE REVIEW

File: src/pages/OAuthRedirector.tsx [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
+import { StoredRouteState } from "types/auth";
 
 export default function OAuthRedirector() {
-  return <Navigate to={localStorage.getItem(OAUTH_PATH_STORAGE_KEY) ?? "/"} />;
+  const retrieved: StoredRouteState | null = (() => {
+    try {
+      const item = localStorage.getItem(OAUTH_PATH_STORAGE_KEY);
+      return item && JSON.parse(item);
+    } catch (_) {
+      return null;
```
Comment (misicnenad) [inline src/pages/OAuthRedirector.tsx:11]:
```suggestion
      return {};
```
nit: return empty object immediately, so there's no need for the null check below.

File: src/pages/Profile/Body/GeneralInfo/DetailsColumn/DetailsColumn.tsx [hunk 2]
```diff
@@ -12,26 +15,42 @@ export default function DetailsColumn({ className = "" }) {
...
-          {isEmpty(active_in_countries)
-            ? p.hq_country
-            : active_in_countries.join(", ")}
-        </Detail>
-        <Tags {...p} />
-        {p.social_media_urls && (
-          <Socials social_media_urls={p.social_media_urls} />
+      <div className={`${className} w-full lg:w-96`}>
+        <div
+          className={`flex flex-col gap-8 w-full p-8 border border-gray-l4 rounded`}
```
Comment (misicnenad) [inline src/pages/Profile/Body/GeneralInfo/DetailsColumn/DetailsColumn.tsx:20]:
```suggestion
          className="flex flex-col gap-8 w-full p-8 border border-gray-l4 rounded"
```

File: src/pages/Profile/Body/GeneralInfo/DetailsColumn/DetailsColumn.tsx [hunk 3]
```diff
@@ -12,26 +15,42 @@ export default function DetailsColumn({ className = "" }) {
...
+              ? p.hq_country
+              : active_in_countries.join(", ")}
+          </Detail>
+          <Tags {...p} />
+          {p.social_media_urls && (
+            <Socials social_media_urls={p.social_media_urls} />
+          )}
+          <DonateButton className="w-full" />
+        </div>
+        {!(p.claimed ?? true) && (
```
Comment (misicnenad) [inline src/pages/Profile/Body/GeneralInfo/DetailsColumn/DetailsColumn.tsx:39]:
```suggestion
        {p.claimed !== false) && (
```
nit: don't have to implement this, it's just an alternative way to write the check

File: src/pages/Registration/Steps/index.tsx [hunk 4]
```diff
@@ -95,3 +114,19 @@ export default function Steps({ classes = "" }: { classes?: string }) {
...
 }
+
+function getClaim(reg: RegistrationState) {
+  /** before documentation, claim intent is determined by initial record */
+  if (reg.step < 4) return reg.data.init.claim;
+
+  /** in documentation and succeeding steps,
+   * registrant might change context from `fresh/new -> claim`.
+   * i.e. inputs org's EIN that happens to be already in our marketplace
+   */
```
Comment (misicnenad) [inline src/pages/Registration/Steps/index.tsx:125]: Love the comments

File: src/types/aws/ap/registration.ts [hunk 5]
```diff
@@ -108,6 +115,7 @@ export type FSASignerDocumentation = {
 export type NonFSADocumentation = {
   DocType: "Non-FSA";
   EIN: string;
+  Claim?: EndowClaim | null;
```
Comment (misicnenad) [inline src/types/aws/ap/registration.ts:118]: Can this be both `null` and `undefined`? Maybe one of them is redundant?

**Review by misicnenad — CHANGES_REQUESTED** [review]
Mainly nits, consider me an approver once addressed.

File: src/types/aws/ap/registration.ts (hunk 5 above)
Comment (ap-justin) [reply src/types/aws/ap/registration.ts:118]: `null` represents `Claim` being reset in the server e.g.
fresh (no claim) -> incidentally inputs unclaimed `EIN` (setting claim) -> enters new EIN instead (resets the `Claim: null`)

File: src/pages/Profile/Body/GeneralInfo/DetailsColumn/DetailsColumn.tsx (hunk 3 above)
Comment (ap-justin) [reply src/pages/Profile/Body/GeneralInfo/DetailsColumn/DetailsColumn.tsx]: fixed https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2822/commits/f94f257a237908843cd6915f312475b4d74a14e3

File: src/pages/Profile/Body/GeneralInfo/DetailsColumn/DetailsColumn.tsx (hunk 2 above)
Comment (ap-justin) [reply src/pages/Profile/Body/GeneralInfo/DetailsColumn/DetailsColumn.tsx]: fixed https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2822/commits/f94f257a237908843cd6915f312475b4d74a14e3

**Review by SovereignAndrey — APPROVED** [review]

**Pull request merged**